### DISCUSSION TITLE
Do not miss any job

### DISCUSSION THREAD

**Discussion by r-priyam** [taskforcesh/bullmq#1502] [Q&A] [answered]
Currently when the application isn't running and if the execution timer comes then the execution is missed after the application starts. Is there anything to prevent it?

For example -

- I add a job to execute after 1 minute
- I stop my application
- Starts the application after 2 minutes
- Job isn't executed.

Any way to tackle this please?

**Comment by manast** [comment, marked as answer]
Please provide test code that reproduces the issue.

**Reply by r-priyam** [reply]
Hello @manast, apologies for the late reply. I was just trying to replicate the issue with this -
```js
import { Queue, Worker } from 'bullmq';

const queue = new Queue('task', { connection: { host: '127.0.0.1', port: 6363 } });
new Worker('task', (job) => console.log(job.data), { connection: { host: '127.0.0.1', port: 6363 } });
// await queue.add('test', { color: 'boo' }, { delay: 10000 });
```

Where I found that it's instant, I mean, whenever the worker is connected, the worker fires the job quickly. This might be the case where I have missed getting the jobs after starting the application the application takes time to load.

Is there any way I can delay the worker job emit by `X` seconds and not instantly when the execution time is passed?

What I mean here is, can the job execution for the expired job be delayed and not execute instantly

**Reply by manast** [reply]
Sorry I do not understand what you mean...

**Reply by r-priyam** [reply]
@manast what I mean is, if an event delay is expired then the worker executes that event as soon as the worker is initialized. Is there any way to delay that? say like I want to start event execution after `X seconds` of worker initialization.

**Reply by manast** [reply]
Then initialize the worker X seconds after.

**Reply by r-priyam** [reply]
Yeah thought so when I didn't find anything specific related to it in the docs, thanks for your time 🙏🏼

**Reply by manast** [reply]
You can pass ```autorun = false``` when instantiating the Worker and then just call ```worker.run()``` when you want: https://api.docs.bullmq.io/interfaces/WorkerOptions.html#autorun
